lib/plug/logger_json.ex

```diff
@@ -149,14 +304,38 @@ defmodule Plug.LoggerJSON do
     Map.drop(log_json, Application.get_env(:plug_logger_json, :suppressed_keys, []))
   end
 
+  @spec format_duration(non_neg_integer(), opts()) :: number()
+  defp format_duration(duration_microseconds, opts) do
+    duration_unit = Keyword.get(opts, :duration_unit, :milliseconds)
+
+    case duration_unit do
+      :nanoseconds ->
+        # Convert microseconds to nanoseconds (multiply by 1000)
+        duration_microseconds * 1000
+
+      :microseconds ->
+        # Already in microseconds
+        duration_microseconds
+
+      :milliseconds ->
+        # Convert microseconds to milliseconds and round to 3 decimal places
+        Float.round(duration_microseconds / 1000, 3)
+
+      _ ->
+        # Default to milliseconds for invalid values
+        Float.round(duration_microseconds / 1000, 3)
+    end
```

Comment:
Could this be done with `System.convert_time_unit`? https://hexdocs.pm/elixir/System.html#convert_time_unit/3